Claims and Role mappings documentation (#191)
epinio/docs: docs/references/authentication_oidc.md

You're now able to login with the credentials in Epinio UI.

## Groups and Roles mapping

> The external identity providers can sometimes provide additional information about the user, for example the groups that he's member of.  

Reviewer: ```suggestion
The external identity providers may provide additional information about the user, for example the groups that he's member of.  
```

These groups can be used to associate a specific role to the user. To do so you need to add a `rolesMapping` key to the `dex-config` secret. The value of the key is a yaml string that will be used to map the groups of a provider to a specific role:

```
rolesMapping: |-
  - connectorId: github
    groups:
    - id: Org1:Admins
      role: admin
    - id: Org1:TeamBlue
      role: user

> config.yaml: |-

Reviewer: What is the purpose of the `config.yaml` key ? It is not described in the preceding text, nor after.

Author: It's the Dex configuration itself. At the beginning it was the only key in this secret, so it was straightforward. The we started adding more keys related on how Epinio integrates with Dex. Should I add something at the beginning maybe?

Line 23: https://github.com/epinio/docs/pull/191/files#diff-88a112af3f97d3ebffec4f38eede241145a6b74d7495746edec9caf36beae58dR23

Reviewer: While the referenced line, i.e.

> To add more connectors, you can edit the Dex config file deployed with the `dex-config` secret

mentions the config file it is not fully clear that this "file" is the `config.yaml` key in the secret.
I would make that link a bit clearer.

Reviewer: And when I said "preceding text" I only had looked at the text in the diff, not the entire file. That was my fault.

  connectors:
  - type: github
    id: github
    name: GitHub
    config:
      loadAllGroups: true
      orgs:
      - name: Org1
```

> The groups will be evaluated in order. In the previous example if the user is a member of both the `Org1:Admins` and `Org1:TeamBlue` then he will get the `admin` role.

Reviewer: ```suggestion
The groups will be evaluated in order. In the previous example if the user is a member of both `Org1:Admins` and `Org1:TeamBlue` then that user will get the `admin` role.
```

> Please note that the role is bound when the user is created, and any modification of the groups will not reflect to already existing users.

Reviewer: ```suggestion
Please note that the role is bound when the user is created. Any modification of the groups will not propagate to already existing users.
```
